src/main/java/com/example/connectfour/service/GameService.java

```diff
@@ -0,0 +1,127 @@
+package com.example.connectfour.service;
+
+import com.example.connectfour.model.Game;
+import org.springframework.stereotype.Service;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class GameService {
+     private final Map<String, Game> games = new ConcurrentHashMap<>();
+
+    public Game createGame() {
+        String gameId = UUID.randomUUID().toString();
+        Game game = new Game(gameId);
+        games.put(gameId, game);
+        return game;
+    }
+
+    public Game getGame(String gameId) {
+        return games.get(gameId);
+    }
+
+    public Game joinGame(String gameId, String playerId) {
+        Game game = games.get(gameId);
+        if (game == null) return null;
+
+        if (game.getPlayer1Id() == null) {
+            game.setPlayer1Id(playerId);
+        } else if (game.getPlayer2Id() == null) {
+            game.setPlayer2Id(playerId);
+            game.setStatus("playing");
+        }
+        return game;
+    }
+
+    public Game makeMove(String gameId, int column, String playerId) {
+        Game game = games.get(gameId);
+        if (game == null || !game.getStatus().equals("playing")) {
+            return null;
+        }
+
+        // Verify it's the player's turn
+        if ((game.getCurrentPlayer() == 1 && !playerId.equals(game.getPlayer1Id())) ||
+            (game.getCurrentPlayer() == 2 && !playerId.equals(game.getPlayer2Id()))) {
+            return null;
+        }
+
+        // Find the lowest empty row in the column
+        int row = -1;
+        for (int r = 5; r >= 0; r--) {
```

Comment:
Could you go with an alternative on this? For loops are very much not in the zeitgeist these days. 
Reduce, map, and filter are

Reply:
+ even if you keep it as a loop. I'd recommend a separate private method for this

Reply:
```
int rows = game.getBoard().length; // total rows
int row = IntStream.iterate(rows - 1, i -> i - 1)
                   .limit(rows)
                   .filter(r -> game.getBoard()[r][column] == 0)
                   .findFirst()
                   .orElse(-1);
```